### PR TITLE
Welcome page

### PR DESCRIPTION
**Pull request by LawrenceWhalen**
## Changes
- Added catches in poros if api limit exceeded
- Added welcome page with links

## Special review requests
- Please pull and check the poro catches on your end.

## Checklist
- [x] Tests written where required
- [x] All tests passing
- [x] SimpleCov at 100%

closes #

### CODE REVIEW

File: app/views/admin/welcome.html.erb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<h2><%= link_to( 'Admin Index', admin_index_path, method: :get, local: true) %><br><br></h2>
+
+<table>
+  <th>Merchant ID:</th>
+  <th>Merchant Name:</th>
+  <% @merchants.each do |merchant| %>
+    <tr><td><%= link_to( "Merchant ##{merchant.id}", merchant_dashboard_path(merchant.id), method: :get, local: true) %></td>
+        <td><%= merchant.name %></td></tr>
+  <% end %>
+</table>
```
Comment (aliyamerali) [inline app/views/admin/welcome.html.erb:10]: 👍  thanks for creating this welcome page

File: spec/poros/github_repo_spec.rb [hunk 2]
```diff
@@ -9,5 +9,13 @@
 
       expect(GithubRepo.name_info).to eq('little-esty-shop')
     end
+    it 'returns an error if the api limit is exceeded' do
+      allow(GithubService).to receive(:repo_info).and_return({
+          message: 'error',
+          name: 'little-esty-shop'
+      })
+
+      expect(GithubRepo.name_info).to eq("API rate limit exceeded.")
```
Comment (aliyamerali) [inline spec/poros/github_repo_spec.rb:18]: 👍 thanks for building tests for this sad path!

**Review by aliyamerali — APPROVED** [review]
@LawrenceWhalen I made a few changes and pushed them to the branch here - when I actually hit the limit live, it turns out the error message was being returned in a single hash, not an array, so the conditions in the if statement were breaking. I updated the conditions to work with this new info, and the tests accordingly. Let me know if it's helpful to jump on a call to discuss!

File: app/poros/github_contributors.rb [hunk 3]
```diff
@@ -2,11 +2,14 @@ class GithubContributors
...
-      if [26797256, 78388882, 5446926, 78574189].include?(contributor[:id])
-        hash[contributor[:id]] = {name: contributor[:login], contributions: contributor[:contributions]}
+    if github_contributor_data.is_a? Array
+      github_contributor_data.each_with_object({}) do |contributor, hash|
+        if [26797256, 78388882, 5446926, 78574189].include?(contributor[:id])
+          hash[contributor[:id]] = {name: contributor[:login], contributions: contributor[:contributions]}
+        end
       end
+    else
+      [["error", { name:"API rate limit exceeded.", contributions: ''}]]
```
Comment (mkrumholz) [inline app/poros/github_contributors.rb]: 😍

File: app/poros/github_pull_requests.rb [hunk 4]
```diff
@@ -1,9 +1,13 @@
...
   def self.merged_pulls
-    GithubService.closed_pulls.count do |pr|
-      !pr[:merged_at].nil?
+    possible_merged = GithubService.closed_pulls
+    if possible_merged.is_a? Array
+      possible_merged.count do |pr|
+        !pr[:merged_at].nil?
+      end
+    else
+      "API rate limit exceeded."
```
Comment (mkrumholz) [inline app/poros/github_pull_requests.rb]: Thanks for adding these in!

File: spec/poros/github_contributors_spec.rb [hunk 5]
```diff
@@ -20,6 +20,17 @@
...
+      it 'returns an error hash if api limit exceeded' do
+        allow(GithubService).to receive(:contributors_info).and_return(
+          {
+          :message=>"API rate limit exceeded for 98.38.149.214. (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.)",
+          :documentation_url=>"https://docs.github.com/rest/overview/resources-in-the-rest-api#rate-limiting"}
+        )
+
+        expect(GithubContributors.contributors_info[0][1][:name]).to eq("API rate limit exceeded.")
+        expect(GithubContributors.contributors_info[0][1][:contributions]).to eq('')
+      end
```
Comment (mkrumholz) [inline spec/poros/github_contributors_spec.rb:33]: Super nice work testing this and getting the sad path in there @LawrenceWhalen !

**Pull request merged**